gdal/ogr/ogrgeometry.cpp

```diff
@@ -1919,7 +1931,7 @@ OGRBoolean
 OGRGeometry::IsValid(  ) const
 
 {
-    if (EQUAL(this->getGeometryName(), "TRIANGLE"))
+    if (EQUAL(this->getGeometryName(), "TRIANGLE") || EQUAL(this->getGeometryName(), "POLYHEDRALSURFACE"))
```

Comment:
I guess that for all those methods where you need to test if it is a non-GEOS/SFCGAL-only geometry, you need a IsSFCGALOnlyGeometry() method to avoid duplicating that test. And it would be probably more efficient to test on the wkbFlatten(getGeometryType()) return than on strings